### PR TITLE
Update SDK

### PR DESCRIPTION
**Pull request by captainsafia**

### CODE REVIEW

**Comment by captainsafia** [comment]
> src/Http/Routing/test/UnitTests/RoutingMetricsTests.cs(144,32): error TBD: (NETCORE_ENGINEERING_TELEMETRY=Build) 'Microsoft.Extensions.Telemetry.Testing.Metering.CollectedMeasurement<long>' is for evaluation purposes only and is subject to change or removal in future updates. (https://aka.ms/dotnet-extensions-warnings/TBD)

This error in `RoutingMetricTests` looks like a result of https://github.com/dotnet/extensions/pull/4130. It seems sufficient to suppress this diagnostic given its in test.

**Review by javiercn — APPROVED** [review]

File: Directory.Build.props [hunk 1]
```diff
@@ -126,6 +126,9 @@
 
     <!-- don't warn about unnecessary trim warning suppressions. can be removed with preview 6. -->
     <NoWarn>$(NoWarn);IL2121</NoWarn>
+
+    <!-- // Temporary diagnostic code from https://github.com/dotnet/extensions/pull/4130 for metrics APIs used in test. -->
+    <NoWarn>TBD</NoWarn>
```
Comment (wtgodbe) [inline Directory.Build.props:131]:
```suggestion
    <NoWarn>$(NoWarn);TBD</NoWarn>
```

Otherwise this will overwrite the existing NoWarn

**Review by wtgodbe — CHANGES_REQUESTED** [review]
Don't want to overwrite existing `NoWarn`

**Review by wtgodbe — APPROVED** [review]

**Comment by RussKie** [comment]
> > src/Http/Routing/test/UnitTests/RoutingMetricsTests.cs(144,32): error TBD: (NETCORE_ENGINEERING_TELEMETRY=Build) 'Microsoft.Extensions.Telemetry.Testing.Metering.CollectedMeasurement' is for evaluation purposes only and is subject to change or removal in future updates. ([aka.ms/dotnet-extensions-warnings/TBD](https://aka.ms/dotnet-extensions-warnings/TBD))
> 
> This error in `RoutingMetricTests` and others looks like a result of [dotnet/extensions#4130](https://github.com/dotnet/extensions/pull/4130). It seems sufficient to suppress this diagnostic given its in test. It's not clear to me how to suppress given it hasn't been designated a diagnostic ID yet...
> 
> Update: it looks like it is feasible by passing the `TBD` diagnostic ID.
> 
> cc: @RussKie Igor Velikorossov FTE @joperezr Jose Perez Rodriguez FTE Do we expect a more permanent diagnostic ID to land soon for this? Also, is suppressing the diagnostic in test code is the right approach?

Yes, it should be safe to suppress. This is still WIP (https://github.com/dotnet/extensions/issues/4161). Eventually, there will be IDs defined.

/cc: @geeknoid

**Pull request merged**